### DISCUSSION TITLE
macOS: crash when running `head` on binary file

### DISCUSSION THREAD

**Discussion by aeons** [ghostty-org/ghostty#2884] [Q&A] [answered]
I have a `pg_dump --format=c` database dump file at 8.4 MB.

If I run `head -n 10 db.dump`, I get the first lines (which happen to be text).
Same for `-n 100`.

With `-n 500`, I start getting to the binary part of the file. Most is printed out, but some command characters are left in the prompt:

![Image](https://github.com/user-attachments/assets/70d97042-e51b-4266-94fc-0abeb8bb9571)

With `-n 800` it's a straight crash to desktop.

With `-n 1000` the application becomes unresponsive (beach balling) and needs to be force quit.

Running `cat` on the file is also crashes immediately.

**Comment by mitchellh** [comment]
Converted to a discussion since as documented in the contribution guidelines we don’t promote to issues until we have an independent reproduction.

Are you able to share the 500 bytes of the file that cause the crash? Or a debug build stack trace?

**Reply by aeons** [reply]
Ah sorry, only read the main readme :/

Will see if I can get a file that reproduces.

**Reply by aeons** [reply, marked as answer]
I have not been able to reproduce this with dummy data. The original database has changed since then, so I can't even recreate the dump that triggers this. I'll close here and reopen if I see it again.
